Source/OrchBase/OrchBaseLink.cpp

```diff
@@ -144,7 +144,178 @@ if (pT->linked) {                      // Task already linked?
 
 */
 // Place and set the link flag if the placement was successful.
-if (!pPlace->Place(pT)) pT->LinkFlag();
+// if (!pPlace->Place(pT)) pT->LinkFlag();
+if (!pPlace->Place(pT))
+{
+   // send data to NameServer. Once the P_builder is threaded, this
+   // should be moved to the build command.
+   int nsRank = Q::NAP;
+   MPI_Comm nsComm = MPI_COMM_NULL;
+   vector<pair<unsigned, int> >superRanks;
+   for (unsigned comm = 0; comm < Comms.size(); comm++)

```

Comment:
This particular iteration is different from line 333, OrchBaseTask.cpp, because it searches for Motherships: I want to pair Motherships and NameServers on the same comm. Lines 333-353 are duplicated once in OrchBaseTask.cpp - which didn't seem like enough times to really justify a separate method for a basic and very context-specific loop. I am purposefully duplicating the code because this fragment generates 2 used values and for such a simple piece I would rather avoid methods whose functionality is achieved via side effects.

If it is thought desirable notwithstanding, I don't fundamentally care how it's done, but I personally think that too many very minor methods simply to avoid a small number of duplicate lines drifts towards over-atomisation of methods, which makes the code hard to read, because then you have to go searching sometimes indeed in separate translation functions for trivial little utilities that only achieve something very specific used in a very narrow context. 

A stronger argument could be made for condensing all the search-through-comms looking for a process calls, I don't know how many of these there are, but there might be more, if this could be made general to look for any given process, I think it might add more value, but again that depends upon how often such a method would actually be called. 

I emphasise again that I don't fundamentally care, if you are insistent upon this modification then I'll happily do it.